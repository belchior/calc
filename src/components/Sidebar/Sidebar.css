.Sidebar {
  z-index: 1;
  overflow-x: hidden;
  display: flex;
  flex-flow: column nowrap;
  flex: 0 1 15em;
  font-family: 'Montserrat Alternates';
  background: linear-gradient(to bottom, #f0f0f0 6em, #e6e6e6 6em);
  box-shadow: 0 0 3px rgba(0,0,0,0.4);
  transition: width 0s linear 0.2s, background-color 0.2s linear;
}

.Sidebar .author {
  position: relative;
  box-sizing: border-box;
  padding: 3em 0 2em 0;
  text-align: center;
}
.Sidebar .author-avatar {
  position: relative;
  display: block;
  margin: 0 auto 1em auto;
  border-radius: 50%;
  width: 5em;
  height: 5em;
}
.Sidebar .author-name {
  display: block;
  font-size: 1.8em;
  line-height: 1.2em;
}

.Sidebar .list {
  flex: 1 1 100%;
}
.Sidebar .list-item {
  box-sizing: border-box;
  display: block;
  padding: 0.5em;
}
.Sidebar .list-item-image {
  display: inline-block;
  margin: 0 0.5em;
  width: 2em;
  height: 2em;
}
.Sidebar .link-custom {
  margin: 0 0 0.5em 0;
  background: #b48fe6;
  color: #ffffff;
}
.Sidebar .link-terminal {
  margin: 0 0 0.5em 0;
  background: #66cc66;
  color: #ffffff;
}
.Sidebar .link-macwidget {
  background: #85acd3;
  color: #ffffff;
}
.Sidebar .link-github {
  text-align: center;
}
.Sidebar .link-github .list-item-image {
  margin: 0 0.5em 0 0;
}

.Sidebar .external-links {
  flex: 0 1;
  margin: 0 0 0.5em 0;
}

@media (max-width: 46em) {
  .Sidebar {
    flex-flow: row;
    flex: 0 1 auto;
    height: 4em;
    order: 1;
  }
  .Sidebar .author {
    padding: 0;
  }
  .Sidebar .author-avatar {
    display: inline-block;
    margin: 1em;
    width: 2em;
    height: 2em;
  }
  .Sidebar .author-name {
    display: none;
  }
  .Sidebar .list {
    display: flex;
    text-align: center;
  }
  .Sidebar .list-item {
    flex: 1 1 auto;
    margin: 0;
    padding: 1em 0.5em;
  }
  .Sidebar .list-item-text {
    display: none;
  }
  .Sidebar .external-links .list-item {
    padding: 1em;
  }
  .Sidebar .external-links .list-item-image {
    margin: 0;
  }
}
