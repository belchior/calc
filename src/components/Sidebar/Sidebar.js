import React, { Component } from 'react';
import {Link} from 'react-router-dom';
import './Sidebar.css';
import iconCustom from '../../images/icon-custom.svg';
import iconTerminal from '../../images/icon-terminal.svg';
import iconMacwidget from '../../images/icon-macwidget.svg';
import iconGithub from '../../images/icon-github.svg';

class Sidebar extends Component {

  render() {
    return (
      <aside className="Sidebar">
        <a className="author" href="https://github.com/belchior">
          <img className="author-avatar" src="https://avatars0.githubusercontent.com/u/2656585?v=3" alt="Fotografia de Belchior Oliveira" />
          <span className="author-name">Belchior Oliveira</span>
        </a>
        <nav className="list internal-links">
          <Link className="list-item link-custom" to="/calc/">
            <img className="list-item-image" src={iconCustom} alt="Calc Custom" />
            <span className="list-item-text">Calc Custom</span>
          </Link>
          <Link className="list-item link-terminal" to="/calc/terminal">
            <img className="list-item-image" src={iconTerminal} alt="Calc Terminal" />
            <span className="list-item-text">Calc Terminal</span>
          </Link>
          <Link className="list-item link-macwidget" to="/calc/macwidget">
            <img className="list-item-image" src={iconMacwidget} alt="Calc Macwidget" />
            <span className="list-item-text">Calc Macwidget</span>
          </Link>
        </nav>
        <nav className="list external-links">
          <a className="list-item link-github" href="https://github.com/belchior/calc">
            <img className="list-item-image" src={iconGithub} alt="Logotipo do GitHub" />
            <span className="list-item-text">Calc no GitHub</span>
          </a>
        </nav>
      </aside>
    );
  }
}

export default Sidebar;
