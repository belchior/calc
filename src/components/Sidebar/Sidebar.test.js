import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router';
import Sidebar from './index';

test('renders Sidebar without crashing', () => {
  let tree = renderer.create(<MemoryRouter><Sidebar /></MemoryRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
