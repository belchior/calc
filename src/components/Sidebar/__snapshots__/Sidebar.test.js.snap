// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Sidebar without crashing 1`] = `
<aside
  className="Sidebar"
>
  <a
    className="author"
    href="https://github.com/belchior"
  >
    <img
      alt="Fotografia de Belchior Oliveira"
      className="author-avatar"
      src="https://avatars0.githubusercontent.com/u/2656585?v=3"
    />
    <span
      className="author-name"
    >
      Belchior Oliveira
    </span>
  </a>
  <nav
    className="list internal-links"
  >
    <a
      className="list-item link-custom"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Calc Custom"
        className="list-item-image"
        src="icon-custom.svg"
      />
      <span
        className="list-item-text"
      >
        Calc Custom
      </span>
    </a>
    <a
      className="list-item link-terminal"
      href="/terminal"
      onClick={[Function]}
    >
      <img
        alt="Calc Terminal"
        className="list-item-image"
        src="icon-terminal.svg"
      />
      <span
        className="list-item-text"
      >
        Calc Terminal
      </span>
    </a>
    <a
      className="list-item link-macwidget"
      href="/macwidget"
      onClick={[Function]}
    >
      <img
        alt="Calc Macwidget"
        className="list-item-image"
        src="icon-macwidget.svg"
      />
      <span
        className="list-item-text"
      >
        Calc Macwidget
      </span>
    </a>
  </nav>
  <nav
    className="list external-links"
  >
    <a
      className="list-item link-github"
      href="https://github.com/belchior/calc"
    >
      <img
        alt="Logotipo do GitHub"
        className="list-item-image"
        src="icon-github.svg"
      />
      <span
        className="list-item-text"
      >
        Calc no GitHub
      </span>
    </a>
  </nav>
</aside>
`;
